src/blockchain/examples/simple_node.rs

```diff
@@ -108,44 +124,37 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // Listen on all interfaces and whatever port the OS assigns
     swarm.listen_on("/ip4/0.0.0.0/tcp/0".parse()?)?;
 
-    let ed25519_keypair = match id_keys {
-        identity::Keypair::Ed25519(v) => v,
-        identity::Keypair::Rsa(_) => todo!(),
-        identity::Keypair::Secp256k1(_) => todo!(),
-    };
-
-    let mut transactions = vec![];
 
-    storage::append_genesis_block(&filepath, &ed25519_keypair);
 
-    let local_id = HashDigest::new(&block::get_publickey_from_keypair(&ed25519_keypair).encode());
+    let local_id = HashDigest::new(&get_publickey_from_keypair(&ed25519_keypair).encode());
     // Kick it off
     loop {
         tokio::select! {
             line = stdin.next_line() => {
                 let line = line?.expect("stdin closed");
-                let transaction = block::Transaction::new(
-                    block::PartialTransaction::new(
-                        block::TransactionType::Create,
+                let transaction = Transaction::new(
+                    PartialTransaction::new(
+                        TransactionType::Create,
                         local_id,
                         line.as_bytes().to_vec(),
                     ),
                     &ed25519_keypair,
                 );
-                transactions.push(transaction);
-                let (parent_hash, previous_number, previous_commiter) = storage::read_last_block(&filepath);
-
-                if check_number == APART_ONE_COMMIT && previous_commiter == local_id {
-                        println!("The Commit Permission is limited, Please wait others commit");
-                        continue;
-                }
-
-                let block = blockchain::new_block(&ed25519_keypair, &transactions, parent_hash, previous_number);
-                println!("---------");
-                println!("---------");
-                println!("Add a New Block : {:?}", block);
-                swarm.behaviour_mut().floodsub.publish(floodsub_topic.clone(), bincode::serialize(&block).unwrap());
-                storage::write_block(&filepath, block);
+                chain.add_block_listener(move |b: Block| {
+                    println!("---------");
+                    println!("---------");
+                    println!("Add a New Block : {:?}", b);
+                    // must put this in a mux
+                    // write_block(&filepath.clone(),b);
+                }).add_block_listener(move |_b: Block| {
+                    // TODO
+                    // swarm.behaviour_mut().floodsub.publish(floodsub_topic.clone(), bincode::serialize(&block).unwrap());
+                });
```

Comment:
Thinking about this more I feel like we should do

```suggestion
                    // TODO(chb0github): Should be wrapped in mutex
                    // write_block(&filepath.clone(), b);
                })
```

The submit transaction + consensus engine will take care of publishing the blocks soon enough 🤞

Reply:
Sure. I was just migrating this case